### ISSUE TITLE
App crashes after changing the mode and click on an app

### ISSUE THREAD

**Issue by shifoc** [MuntashirAkon/AppManager#722]
- [x] I know what my device, OS and App Manager versions are
- [] I know how to take logs
- [x] I know how to reproduce the issue which may not be specific to my device

**Describe the bug**
App crashes after changing the mode and click on an app. I would also like to add that the app is not connecting successfully to wireless debug, it says wireless debug(inferred no root) 

**Crash logs**
```
java.lang.RuntimeException: Unable to start activity ComponentInfo{io.github.muntashirakon.AppManager/io.github.muntashirakon.AppManager.details.AppDetailsActivity}: java.lang.RuntimeException: Cannot create an instance of class io.github.muntashirakon.AppManager.details.AppDetailsViewModel
    at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4035)
    at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:4201)
    at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:103)
    at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
    at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
    at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2438)
    at android.os.Handler.dispatchMessage(Handler.java:106)
    at android.os.Looper.loopOnce(Looper.java:226)
    at android.os.Looper.loop(Looper.java:313)
    at android.app.ActivityThread.main(ActivityThread.java:8646)
    at java.lang.reflect.Method.invoke(Native Method)
    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:567)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1135)
 Caused by: java.lang.RuntimeException: Cannot create an instance of class io.github.muntashirakon.AppManager.details.AppDetailsViewModel
   at androidx.lifecycle.ViewModelProvider$AndroidViewModelFactory.create(ViewModelProvider.java:275)
   at androidx.lifecycle.SavedStateViewModelFactory.create(SavedStateViewModelFactory.java:112)
   at androidx.lifecycle.ViewModelProvider.get(ViewModelProvider.java:185)
   at androidx.lifecycle.ViewModelProvider.get(ViewModelProvider.java:150)
   at io.github.muntashirakon.AppManager.details.AppDetailsActivity.onAuthenticated(AppDetailsActivity.java:64)
   at io.github.muntashirakon.AppManager.BaseActivity.onCreate(BaseActivity.java:65)
   at android.app.Activity.performCreate(Activity.java:8290)
   at android.app.Activity.performCreate(Activity.java:8270)
   at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1329)
   at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4009)
   at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:4201)
   at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:103)
   at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
   at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
   at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2438)
   at android.os.Handler.dispatchMessage(Handler.java:106)
   at android.os.Looper.loopOnce(Looper.java:226)
   at android.os.Looper.loop(Looper.java:313)
   at android.app.ActivityThread.main(ActivityThread.java:8646)
   at java.lang.reflect.Method.invoke(Native Method)
   at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:567)
   at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1135)

Device Info:
App version: 3.0.0-alpha02
App version code: 401
Android build version: S908EXXU1AVC9
Android release version: 12
Android SDK version: 31
Android build ID: SP1A.210812.016.S908EXXU1AVC9
Device brand: samsung
Device manufacturer: samsung
Device name: b0q
Device model: SM-S908E
Device product name: b0qxxx
Device hardware name: qcom
ABIs: [arm64-v8a, armeabi-v7a, armeabi]
ABIs (32bit): [armeabi-v7a, armeabi]
ABIs (64bit): [arm64-v8a]
System language: en-US
In-App Language: auto
Mode: adb_wifi
```

**Device info**
 - Device: S22 Ultra
 - OS Version: Android 12
 - App Manager Version: 3.0.0-alpha2
 - Mode: wireless debug than I changed it to auto

**Comment by MuntashirAkon** [comment]
For some reason, the crashlog is quite unrelated to the issue. This could be a caching error. Can you reproduce it again so that it can take the correct logs (if this is indeed an error). Otherwise, install a debug build and take some logs.

**Comment by shifoc** [comment]
I reproduced it by changing the mode to wireless debug and tried to click on an app from the list.

java.lang.RuntimeException: Unable to start activity ComponentInfo{io.github.muntashirakon.AppManager/io.github.muntashirakon.AppManager.details.AppDetailsActivity}: java.lang.RuntimeException: Cannot create an instance of class io.github.muntashirakon.AppManager.details.AppDetailsViewModel
    at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4035)
    at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:4201)
    at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:103)
    at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
    at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
    at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2438)
    at android.os.Handler.dispatchMessage(Handler.java:106)
    at android.os.Looper.loopOnce(Looper.java:226)
    at android.os.Looper.loop(Looper.java:313)
    at android.app.ActivityThread.main(ActivityThread.java:8646)
    at java.lang.reflect.Method.invoke(Native Method)
    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:567)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1135)
 Caused by: java.lang.RuntimeException: Cannot create an instance of class io.github.muntashirakon.AppManager.details.AppDetailsViewModel
   at androidx.lifecycle.ViewModelProvider$AndroidViewModelFactory.create(ViewModelProvider.java:275)
   at androidx.lifecycle.SavedStateViewModelFactory.create(SavedStateViewModelFactory.java:112)
   at androidx.lifecycle.ViewModelProvider.get(ViewModelProvider.java:185)
   at androidx.lifecycle.ViewModelProvider.get(ViewModelProvider.java:150)
   at io.github.muntashirakon.AppManager.details.AppDetailsActivity.onAuthenticated(AppDetailsActivity.java:64)
   at io.github.muntashirakon.AppManager.BaseActivity.onCreate(BaseActivity.java:65)
   at android.app.Activity.performCreate(Activity.java:8290)
   at android.app.Activity.performCreate(Activity.java:8270)
   at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1329)
   at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4009)
   at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:4201)
   at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:103)
   at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
   at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
   at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2438)
   at android.os.Handler.dispatchMessage(Handler.java:106)
   at android.os.Looper.loopOnce(Looper.java:226)
   at android.os.Looper.loop(Looper.java:313)
   at android.app.ActivityThread.main(ActivityThread.java:8646)
   at java.lang.reflect.Method.invoke(Native Method)
   at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:567)
   at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1135)

Device Info:
App version: 3.0.0-alpha02
App version code: 401
Android build version: S908EXXU1AVC9
Android release version: 12
Android SDK version: 31
Android build ID: SP1A.210812.016.S908EXXU1AVC9
Device brand: samsung
Device manufacturer: samsung
Device name: b0q
Device model: SM-S908E
Device product name: b0qxxx
Device hardware name: qcom
ABIs: [arm64-v8a, armeabi-v7a, armeabi]
ABIs (32bit): [armeabi-v7a, armeabi]
ABIs (64bit): [arm64-v8a]
System language: en-US
In-App Language: auto
Mode: adb_wifi

**Comment by MuntashirAkon** [comment]
> tried to click on an app from the list.

Ah, I see. Let me see if I can reproduce this issue. In the mean time, you should wait for 10-15 seconds before trying to open a file. It takes a few second to connect to the remote service.

**Comment by MuntashirAkon** [comment]
Try the latest debug (from 8c28f4012ba30950aeb62b6799fc3a79a34dc238) and also take logs during the crash as the crash report is not enough in this case.

**Comment by shifoc** [comment]
```
java.lang.RuntimeException: Unable to start activity ComponentInfo{io.github.muntashirakon.AppManager.debug/io.github.muntashirakon.AppManager.details.AppDetailsActivity}: java.lang.RuntimeException: Cannot create an instance of class io.github.muntashirakon.AppManager.details.AppDetailsViewModel
    at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4035)
    at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:4201)
    at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:103)
    at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
    at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
    at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2438)
    at android.os.Handler.dispatchMessage(Handler.java:106)
    at android.os.Looper.loopOnce(Looper.java:226)
    at android.os.Looper.loop(Looper.java:313)
    at android.app.ActivityThread.main(ActivityThread.java:8663)
    at java.lang.reflect.Method.invoke(Native Method)
    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:567)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1135)
 Caused by: java.lang.RuntimeException: Cannot create an instance of class io.github.muntashirakon.AppManager.details.AppDetailsViewModel
   at androidx.lifecycle.ViewModelProvider$AndroidViewModelFactory.create(ViewModelProvider.java:275)
   at androidx.lifecycle.SavedStateViewModelFactory.create(SavedStateViewModelFactory.java:112)
   at androidx.lifecycle.ViewModelProvider.get(ViewModelProvider.java:185)
   at androidx.lifecycle.ViewModelProvider.get(ViewModelProvider.java:150)
   at io.github.muntashirakon.AppManager.details.AppDetailsActivity.onAuthenticated(AppDetailsActivity.java:64)
   at io.github.muntashirakon.AppManager.BaseActivity.onCreate(BaseActivity.java:65)
   at android.app.Activity.performCreate(Activity.java:8290)
   at android.app.Activity.performCreate(Activity.java:8270)
   at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1329)
   at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4009)
   at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:4201)
   at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:103)
   at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
   at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
   at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2438)
   at android.os.Handler.dispatchMessage(Handler.java:106)
   at android.os.Looper.loopOnce(Looper.java:226)
   at android.os.Looper.loop(Looper.java:313)
   at android.app.ActivityThread.main(ActivityThread.java:8663)
   at java.lang.reflect.Method.invoke(Native Method)
   at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:567)
   at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1135)
```
Device Info:
App version: 3.0.0-alpha03-DEBUG
App version code: 402
Android build version: S908EXXS2AVDD
Android release version: 12
Android SDK version: 31
Android build ID: SP1A.210812.016.S908EXXS2AVDD
Device brand: samsung
Device manufacturer: samsung
Device name: b0q
Device model: SM-S908E
Device product name: b0qxxx
Device hardware name: qcom
ABIs: [arm64-v8a, armeabi-v7a, armeabi]
ABIs (32bit): [armeabi-v7a, armeabi]
ABIs (64bit): [arm64-v8a]
System language: en-US
In-App Language: auto
Mode: adb_wifi



I am sorry I don't know how to take logs, can you tell me how?

**Comment by MuntashirAkon** [comment]
Logs does not include the crashlog itself, meaning they were not taken during the crash. Note that you have to take logs using ADB and not using App Manager since it is crashing. You can run logcat via the ADB command `adb logcat` or `adb shell logcat`. If you can find App Manager process ID, you can also append ` --pid=<pid>` (replace `<pid>` with App Manager's current process ID) in order to filter the logs properly.

**Comment by MuntashirAkon** [comment]
Similar issues in Samsung reported via E-Mail:

```


java.lang.RuntimeException: Unable to start activity
ComponentInfo{io.github.muntashirakon.AppManager/io.github.muntashirakon.AppManager.details.AppDetailsActivity}:
java.lang.RuntimeException: Cannot create an instance of class
io.github.muntashirakon.AppManager.details.AppDetailsViewModel
    at
android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4035)
    at
android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:4201)
    at
android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:103)
    at
android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
    at
android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
    at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2438)
    at android.os.Handler.dispatchMessage(Handler.java:106)
    at android.os.Looper.loopOnce(Looper.java:226)
    at android.os.Looper.loop(Looper.java:313)
    at android.app.ActivityThread.main(ActivityThread.java:8663)
    at java.lang.reflect.Method.invoke(Native Method)
    at
com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:567)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1135)
Caused by: java.lang.RuntimeException: Cannot create an instance of class
io.github.muntashirakon.AppManager.details.AppDetailsViewModel
   at
androidx.lifecycle.ViewModelProvider$AndroidViewModelFactory.create(ViewModelProvider.java:275)
   at
androidx.lifecycle.SavedStateViewModelFactory.create(SavedStateViewModelFactory.java:112)
   at androidx.lifecycle.ViewModelProvider.get(ViewModelProvider.java:185)
   at androidx.lifecycle.ViewModelProvider.get(ViewModelProvider.java:150)
   at
io.github.muntashirakon.AppManager.details.AppDetailsActivity.onAuthenticated(AppDetailsActivity.java:64)
   at
io.github.muntashirakon.AppManager.BaseActivity.onCreate(BaseActivity.java:65)
   at android.app.Activity.performCreate(Activity.java:8290)
   at android.app.Activity.performCreate(Activity.java:8270)
   at
android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1329)
   at
android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4009)
   at
android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:4201)
   at
android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:103)
   at
android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
   at
android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
   at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2438)
   at android.os.Handler.dispatchMessage(Handler.java:106)
   at android.os.Looper.loopOnce(Looper.java:226)
   at android.os.Looper.loop(Looper.java:313)
   at android.app.ActivityThread.main(ActivityThread.java:8663)
   at java.lang.reflect.Method.invoke(Native Method)
   at
com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:567)
   at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1135)

Device Info:
App version: 3.0.0-alpha03
App version code: 402
Android build version: G780GXXS3CVC4
Android release version: 12
Android SDK version: 31
Android build ID: SP1A.210812.016.G780GXXS3CVC4
Device brand: samsung
Device manufacturer: samsung
Device name: r8q
Device model: SM-G780G
Device product name: r8qxx
Device hardware name: qcom
ABIs: [arm64-v8a, armeabi-v7a, armeabi]
ABIs (32bit): [armeabi-v7a, armeabi]
ABIs (64bit): [arm64-v8a]
System language: pt-BR
In-App Language: auto
Mode: adb_tcp
```

**Comment by MuntashirAkon** [comment]
And Xiaomi (reported much earlier):

```
java.lang.RuntimeException: Unable to start activity ComponentInfo{io.github.muntashirakon.AppManager/io.github.muntashirakon.AppManager.details.AppDetailsActivity}: java.lang.RuntimeException: Cannot create an instance of class io.github.muntashirakon.AppManager.details.AppDetailsViewModel
    at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2817)
    at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2895)
    at android.app.ActivityThread.-wrap11(Unknown Source:0)
    at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1616)
    at android.os.Handler.dispatchMessage(Handler.java:106)
    at android.os.Looper.loop(Looper.java:176)
    at android.app.ActivityThread.main(ActivityThread.java:6651)
    at java.lang.reflect.Method.invoke(Native Method)
    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:547)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:824)
 Caused by: java.lang.RuntimeException: Cannot create an instance of class io.github.muntashirakon.AppManager.details.AppDetailsViewModel
   at androidx.lifecycle.ViewModelProvider$AndroidViewModelFactory.create(ViewModelProvider.java:275)
   at androidx.lifecycle.SavedStateViewModelFactory.create(SavedStateViewModelFactory.java:112)
   at androidx.lifecycle.ViewModelProvider.get(ViewModelProvider.java:185)
   at androidx.lifecycle.ViewModelProvider.get(ViewModelProvider.java:150)
   at io.github.muntashirakon.AppManager.details.AppDetailsActivity.onAuthenticated(AppDetailsActivity.java:64)
   at io.github.muntashirakon.AppManager.BaseActivity.onCreate(BaseActivity.java:65)
   at android.app.Activity.performCreate(Activity.java:7088)
   at android.app.Activity.performCreate(Activity.java:7079)
   at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1215)
   at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2770)
   at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2895)
   at android.app.ActivityThread.-wrap11(Unknown Source:0)
   at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1616)
   at android.os.Handler.dispatchMessage(Handler.java:106)
   at android.os.Looper.loop(Looper.java:176)
   at android.app.ActivityThread.main(ActivityThread.java:6651)
   at java.lang.reflect.Method.invoke(Native Method)
   at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:547)
   at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:824)

Device Info:
App version: 3.0.0-alpha02
App version code: 401
Android build version: V11.0.2.0.OCKMIXM
Android release version: 8.1.0
Android SDK version: 27
Android build ID: CUSTOMER_VERSION_riva
Device brand: Xiaomi
Device manufacturer: Xiaomi
Device name: riva
Device model: Redmi 5A
Device product name: riva
Device hardware name: qcom
ABIs: [arm64-v8a, armeabi-v7a, armeabi]
ABIs (32bit): [armeabi-v7a, armeabi]
ABIs (64bit): [arm64-v8a]
System language: en-IN
In-App Language: auto
Mode: adb_tcp
```

**Comment by shifoc** [comment]
[logcatDump123.txt](https://github.com/MuntashirAkon/AppManager/files/8694040/logcatDump123.txt)

**Comment by shifoc** [comment]
I sent the logs by mail because I don't know if they can contain private information

**Comment by MuntashirAkon** [comment]
Thanks. I've figured out the issue.

**Comment by MuntashirAkon** [comment]
Should be fixed in 56231212079be1388c4c72c2d9bada42cc540347
